lib/danger-packwerk/danger_package_todo_yml_changes.rb

```diff
@@ -29,22 +29,26 @@ class DangerPackageTodoYmlChanges < Danger::Plugin
         offenses_formatter: T.nilable(Update::OffensesFormatter),
         before_comment: BeforeComment,
         max_comments: Integer,
-        violation_types: T::Array[String]
+        violation_types: T::Array[String],
+        base_root_directory: String
       ).void
     end
     def check(
       offenses_formatter: nil,
       before_comment: DEFAULT_BEFORE_COMMENT,
       max_comments: DEFAULT_MAX_COMMENTS,
-      violation_types: DEFAULT_VIOLATION_TYPES
+      violation_types: DEFAULT_VIOLATION_TYPES,
+      base_root_directory: nil
     )
       offenses_formatter ||= Update::DefaultFormatter.new
       repo_link = github.pr_json[:base][:repo][:html_url]
       org_name = github.pr_json[:base][:repo][:owner][:login]
 
-      changed_package_todo_ymls = (git.modified_files + git.added_files + git.deleted_files).grep(PACKAGE_TODO_PATTERN)
+      # TODO: write explanation here
+      git_filesystem = Private::GitFilesystem.new(git: git, root: base_root_directory || '')
+      changed_package_todo_ymls = (git_filesystem.modified_files + git_filesystem.added_files + git_filesystem.deleted_files).grep(PACKAGE_TODO_PATTERN)
 
-      violation_diff = get_violation_diff(violation_types)
+      violation_diff = get_violation_diff(violation_types, base_root_directory: base_root_directory)
```

Comment:
Can we just pass in the `git_filesystem` here instead of constructing a new one inside the function?

Reply:
Yep, definitely took that approach first, but `get_violation_diff` is a publicly exposed function, so we need this to maintain backwards compatibility. Adding `git_filesystem` as an optional param is a possibility, but I don't think we want to bubble it all the way up to end users